api/geth_backend.go

```diff
@@ -1513,14 +1646,40 @@ func (b *GethStatusBackend) prepareSubAccounts(mnemonic, keyUID string, derivedA
 	return []*accounts.Account{walletAccount, chatAccount}, nil
 }
 
-func (b *GethStatusBackend) getDerivedAddresses(id string) (map[string]generator.AccountInfo, error) {
-	accountGenerator := b.accountManager.AccountsGenerator()
-	derivedAddresses, err := accountGenerator.DeriveAddresses(id, paths)
+func (b *GethStatusBackend) prepareForKeycard(request *requests.CreateAccount, input *prepareAccountInput, response *accountBundle) (*accountBundle, error) {
+	if request.KeycardInstanceUID == "" {
+		return response, nil
+	}
+
+	kp := wallet.NewKeycardPairings()
+	kp.SetKeycardPairingsFile(response.nodeConfig.KeycardPairingDataFile)
+	pairings, err := kp.GetPairings()
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "failed to get keycard pairings")
+	}
+
+	keycard, ok := pairings[request.KeycardInstanceUID]
+	if !ok {
+		return nil, errors.New("keycard not found in pairings file")
+	}
+
+	response.settings.KeycardInstanceUID = request.KeycardInstanceUID
```

Comment:
@igor-sirotin I am not sure that we need any of `KeycardInstanceUID`, `KeycardPairedOn` and `KeycardPairing`? We most likely need to remove them, to pollute the code anymore.

If not here maybe you can create a new issue for removing them, wdyt?

Reply:
But... maybe the mobile app is using them somehow?!? But even so, they will switch to the new keycard lib (soon) and then we can remove them if not now.

Reply:
@saledjenic oh, I didn't think that they were added to `settings` just to create the account. Was this the intention?

> maybe the mobile app is using them somehow?!?

Since mobile doesn't support keycard right now, it should be safe to only think of desktop.

Reply:
>  Was this the intention?

I think that's there from the first keycard implementation which was far before we did it in the desktop app, developed by mobile devs. If they are not using it, we should be free to clean than part from the db and code.

Reply:
thanks @qfrank yes i'm working on this atm, in v2 we'll have support only for logging in with keycard, so basically on mobile only `.loginWithKeycard` will be used